### PR TITLE
Implement seg v3

### PR DESCRIPTION
**Pull request by avv-va**

### CODE REVIEW

File: ultralytics/nn/tasks.py [hunk 1]
```diff
@@ -375,8 +382,105 @@ def __init__(self, cfg='yolov8n-pose-seg.yaml', ch=3, nc=None, data_kpt_shape=(N
...
+        """
+        if not hasattr(self, 'criterion'):
+            self.criterion = self.init_criterion()
+        
+        if preds:
+            assert self.training is False
+            box_kpt_loss = self.calc_box_kpt_loss(preds=preds, batch=batch)
+            seg_cls_loss = self.calc_seg_cls_loss(preds=preds, batch=batch)
+            seg_obj_loss_item = torch.Tensor([0]).to(seg_cls_loss[0].device) # To avoid running into errors
+            loss_sum = box_kpt_loss[0] + seg_cls_loss[0]
```
[the remark this answers is not in the document]
Comment (avv-va) [reply ultralytics/nn/tasks.py:436]: Not sure if I got your question right but loss functions output a tuple of (summed_loss (integer), [loss_item1, loss_item2, ... ]).
